gateway/src/cluster/impl.rs

```diff
@@ -13,11 +10,80 @@ use futures_util::{
 };
 use std::{
     collections::HashMap,
+    error::Error,
+    fmt::{Display, Formatter, Result as FmtResult},
     iter::FromIterator,
     sync::{Arc, Weak},
 };
+use twilight_http::Error as HttpError;
 use twilight_model::gateway::event::Event;
 
+/// Sending a command to a shard failed.
+#[derive(Debug)]
+pub enum ClusterCommandError {
+    /// The shard exists, but sending the provided value failed.
+    Sending {
+        /// Reason for the error.
+        source: CommandError,
+    },
+    /// The provided shard ID does not exist.
+    ShardNonexistent {
+        /// The provided shard ID.
+        id: u64,
+    },
+}
+
+impl Display for ClusterCommandError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        match self {
+            Self::Sending { source } => Display::fmt(source, f),
+            Self::ShardNonexistent { id } => {
+                f.write_fmt(format_args!("shard {} does not exist", id,))
+            }
+        }
+    }
+}
+
+impl Error for ClusterCommandError {
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
+        match self {
+            Self::Sending { source } => Some(source),
+            Self::ShardNonexistent { .. } => None,
+        }
+    }
+}
+
+/// Starting a cluster failed.
+#[derive(Debug)]
+pub enum ClusterStartError {
+    /// Retrieving the bot's gateway information via the HTTP API failed.
+    ///
+    /// This information includes the number of shards for the cluster to
```

Comment:
Can this comment be clarified? What info does this end up containing? At a cursory glance it looks to be from the HTTP client.

Reply:
addressed